### PR TITLE
Added option to add a custom annotation, clarified multiQC results and fixed issues

### PR DESCRIPTION
**Pull request by svarona**
Included [issue #390](https://github.com/nf-core/viralrecon/issues/390), issue #336 enhancements and issue https://github.com/nf-core/viralrecon/issues/419
- Added new param to add a custom annotation (gtf or gff format)
- Added new subworkflow to perform the additional annotation
- Allowed SNPEFF_BUILD process to detect wether annotation is in gtf or gff format
- Fixed variants_long_table so it won't select first annotation feature only, but remove the features that are not `upstream` or `downstream`
- Added a clearly description of variants_long_table formats and content for both nanopore and illumina.
- Clarify #SNPs and #INDELS in the illumina's multiqc

## PR checklist

- [X] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/viralrecon/tree/master/.github/CONTRIBUTING.md)
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [x] Usage Documentation in `docs/usage.md` is updated.
- [x] Output Documentation in `docs/output.md` is updated.
- [X] `CHANGELOG.md` is updated.
- [x] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -83,6 +84,10 @@ jobs:
           wget -qO- get.nextflow.io | bash
           sudo mv nextflow /usr/local/bin/
 
+      - name: Download GTF for additional annotation
+        run: |
+          wget https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/009/858/895/GCA_009858895.3_ASM985889v3/GCA_009858895.3_ASM985889v3_genomic.gtf.gz
+
```
Comment (maxulysse) [inline .github/workflows/ci.yml]: If I'm not misunderstanding, you're downloading this every time in this matrix.
Can you add an if statement on the params for this step?

**Review by maxulysse — COMMENTED** [review]
Superb work, I'm not a super fan of the name for this new params: `--additional_annot`
Why not `--additional_annotation`, it's not that much longer.

File: nextflow_schema.json [hunk 2]
```diff
@@ -78,6 +78,14 @@
                     "description": "Full path to GFF annotation file.",
                     "fa_icon": "fas fa-file-invoice"
                 },
+                "additional_annot": {
+                    "type": "string",
+                    "format": "file-path",
+                    "mimetype": "text/plain",
+                    "pattern": "^\\S+(\\.gff|\\.gtf)(\\.gz)?$",
+                    "description": "Full path to additional annotation file in GTF or GFF format.",
```
Comment (maxulysse) [inline nextflow_schema.json:87]: You could mention the optional `.gz`.
I assume there's no need to decompress in such case?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (svarona) [reply .github/workflows/ci.yml]: Sure! I also added it in nanopore when it downloads the medaka model. Thanks for the suggestion!

**Comment by svarona** [comment]
> Superb work, I'm not a super fan of the name for this new params: `--additional_annot` Why not `--additional_annotation`, it's not that much longer.

Done! :heavy_check_mark:

File: nextflow_schema.json (hunk 2 above)
Comment (svarona) [reply nextflow_schema.json:87]: It is the same format as the [--gff param](https://github.com/nf-core/viralrecon/blob/1883a68346f87cc2534323f8dbeceec554a7edc2/nextflow_schema.json#L74)

Should I add the .gz in the description for both? Both of them have a `GUNZIP_GFF` process in the case they are compressed.

**Review by maxulysse — APPROVED** [review]

**Pull request merged**